src/useForm.ts

```diff
@@ -165,14 +166,43 @@ export class FormStore {
     return cache;
   };
 
-  private getFieldsValue = (nameList?: NamePath[]) => {
+  private getFieldsValue = (
+    nameList?: NamePath[],
+    filterFunc?: (meta: Meta) => boolean,
+  ) => {
     this.warningUnhooked();
 
-    if (!nameList) {
+    if (!nameList && !filterFunc) {
       return this.store;
     }
+    if (!filterFunc) {
+      return cloneByNamePathList(this.store, nameList.map(getNamePath));
+    }
+
+    let fieldEntities = this.getFieldEntities(true);
```

Comment:
Move  `this.getFieldEntities(true)` into follow if condition since `pure` mode will cost additional performance effort.